lib/wibox/widget/base.lua

```diff
@@ -321,6 +330,372 @@ function base.place_widget_at(widget, x, y, width, height)
+--                    align  = "center",
+--                    widget = wibox.widget.textbox
+--                },
+--                bg     = "#ff0000",
+--                widget = wibox.widget.background,
+--            },
+--            width    = 300,
+--            strategy = "min",
+--            layout   = wibox.layout.constraint
+--        },
+--        {
+--            -- Add a border around the background
+--            {
+--                {
+--                    markup = "Foobar",
+--                    widget = wibox.widget.textbox
+--                },
+--                bg     = "#0000ff",
+--                widget = wibox.widget.background
+--            },
+--            left   = 10,
+--            right  = 10,
+--            top    = 1,
+--            bottom = 2,
+--            layout = wibox.layout.margin
+--        },
+--        layout = wibox.layout.fixed.horizontal,
+--    }
+--    
+--
+-- Result:
+-- ![Example2 screenshot](../../docs/images/widgetlayout1.png)
+--
+--
+-- **Example 3 :** Use an `wibox.layout.align` latout
+-- The `wibox.layout.align` is a little different. While most layouts will
+-- ignore any `nil` lines, the `align` layout rely on them so `left`, `middle`
+-- and `right` can be defined
+--
+-- Code:
+--    
+--    mywibox[s] : setup {
+--        my_textbox1, -- Left
+--        nil,         -- Nothing in the middle
+--        my_textbox2, -- Right
+--        layout = wibox.layout.fixed.horizontal,
+--    }
+--    
+--
+--
+-- **Example 4 :** Define new widgets
+--
+-- New trivial widgets can be created directly in the layout declaration. Here
+-- is a simple circle widget:
+--
+-- Code:
+--    
+--    mywibox[s] : setup {
+--        fit    = function(self, context, width, height)
+--            return height, height -- A square taking the full height
+--        end,
+--        draw   = function(self, context, cr, width, height)
+--            cr:set_source_rgb(1, 0, 0) -- Red
+--            cr:arc(height/2, height/2, height/2, 0, math.pi*2)
+--            cr:fill()
+--        end,
+--        layout = wibox.widget.base.make_widget,
+--    }
+--    
+-- Result:
+-- ![Example4 screenshot](../../docs/images/widgetlayout2.png)
+--
+-- For more information about how to draw widgets, refer to the `Cairo` api:
+--
+-- * [Path](http://cairographics.org/manual/cairo-Paths.html)
+-- * [Context](http://cairographics.org/manual/cairo-cairo-t.html)
+-- * [Pattern](http://cairographics.org/manual/cairo-cairo-pattern-t.html)
+-- * [transformation](http://cairographics.org/manual/cairo-Transformations.html)
+-- * [Operator](http://cairographics.org/operators/)
+-- * [Pango text](https://developer.gnome.org/pango/stable/)
+--
+--
+-- **Example 5 :** Externally defined widgets and layouts
+--
+-- The declarative system can also use external widgets. This is useful when the
+-- widget is provided by an external module or require complex manupilations
+-- would make the declaration unreadable.
```

Comment:
2 of Elv13 weakness

 * Making stupid mistakes when updating PR at 4 AM
 * Being totally reliant on spellcheck and not having one in its editor